class.php

```diff
@@ -131,13 +137,15 @@ protected function _generatePagesByLanguage(string $baseUrl, string $languageCod
   {
     foreach ($this->_pages->keys() as $key) {
       $page = $this->_pages->$key;
-      $this->_setPageLanguage($page, $languageCode);
-      $path = str_replace($this->_originalBaseUrl, '/', $page->url());
-      $path = $this->_cleanPath($this->_outputFolder . $path . '/index.html');
-      try {
-        $this->_generatePage($page, $path, $baseUrl);
-      } catch (ErrorException $error) {
-        $this->_handleRenderError($error, $key, $languageCode);
+      if(!$this->_ignoreUntranslatedPages || $page->translation($languageCode)->exists()){
+        $this->_setPageLanguage($page, $languageCode);
+        $path = str_replace($this->_originalBaseUrl, '/', $page->url());
+        $path = $this->_cleanPath($this->_outputFolder . $path . '/index.html');
+        try {
+          $this->_generatePage($page, $path, $baseUrl);
+        } catch (ErrorException $error) {
+          $this->_handleRenderError($error, $key, $languageCode);
+        }
```

Comment:
By switching the if-condition we can avoid the code indent:

```suggestion
      if ($this->_ignoreUntranslatedPages && !$page->translation($languageCode)->exists()) {
        return;
      }

      $this->_setPageLanguage($page, $languageCode);
      $path = str_replace($this->_originalBaseUrl, '/', $page->url());
      $path = $this->_cleanPath($this->_outputFolder . $path . '/index.html');
      try {
        $this->_generatePage($page, $path, $baseUrl);
      } catch (ErrorException $error) {
        $this->_handleRenderError($error, $key, $languageCode);
```